EndoStation 2 Electric Boogaloo

Blueturbo47 opened the pull request:
## About The Pull Request

Adds EndoStation a new map for Beestation.
credit to @qwertyquerty for the name

Original pr - https://github.com/BeeStation/BeeStation-Hornet/pull/2925

Me-
I will be trying to be the one who map fixes and the such for EndoStation, I'm still quite new to this so it might take me a bit but I didn't want crusader's map to go to waste.

Also because I'm bad at coding I accidentally included [gh#3115](https://github.com/BeeStation/BeeStation-Hornet/pull/3115). So this pr if merged before 3115 this will be closing it.

## Why It's Good For The Game

Lastcrusader105 -

Adds a new station
for you MRPersor HRPers
what is EndoStation how was it made etc?
EndoStation was made in a joint effort between Dr.Octagon and Nanotrasen after he convinced them being a plasma man he knew all about plasma and would make them both rich, then rumors started about Octagons weird obsession with gene splicing and creating monsters, and the station started to lose funding, Octagon is believed dead

hopefully, you get the feel of this once vibrant station has lost some of its glamour and looks a bit ran down, science was first to be underfunded originally meant to show off Nanotrasen power and reach, at least that's what Octagon told them, Its crew entertainment sector is reached soon as you get off the shuttle each department has a shuttle, one of Nanotrasens failed visions (future-proofing map for away missions like sciences)

Me-
No reason not to have a new map.

## Changelog
:cl: Lastcrusader105, Blueturbo47
add: new station EndoStation
add: more pods on Centcom, an air alarm for plasma rooms and a new pet a gronola Jammor IX, 2 new areas: bridge checkpoints
imageadd: added some area images 2 bridge checkpoints and one for AFT solar control
code: added, plumbing and morgue to mail system, adds new solar control area, adds a plasma alarm (good for plasmeme rooms)
/:cl:

Commit:
Adds Zulu Station

Commit:
Adds 2 new docking ports

Commit:
adds Zulu Station

Commit:
Adds Plumbing and Detectives office to mailroom

Commit:
Adds another Solar:AFT and 2 sec checkpoints:Bridge port+Star

Commit:
Adds Plasma Airalarm

good for plasma rooms

Commit:
Adds a Plasma breath Gondola, Jammor IX

Commit:
Adds ZuluStation

Commit:
adds stuff

Commit:
adds service shuttle

Commit:
added stuff

Commit:
more lights

Commit:
changed door names and access plus more pipes etc

Commit:
adds cameras

Commit:
Adds ZuluStation

Commit:
added spawn points, sorted out "near space" + few areas i missed

Commit:
Update code/modules/mob/living/simple_animal/friendly/gondola.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

Commit:
fixes bacons issues eg double walls etc

Commit:
Merge remote-tracking branch 'origin/ZuluStation' into ZuluStation

Commit:
fixes bacons issues 1-13

Commit:
solves travis bitching

thank to mark for the assist

Commit:
finally sorts Travis

turns out its nitrogen not n2 and oxygen not o2

Commit:
fixes various issues, fixes maint

Commit:
adds Plasma alarm, fixes wording issues

good for plasmaman rooms

Commit:
sorts air alarm out

Commit:
adds Better Maints and lights

Commit:
Adds EndoStation

Commit:
added new areas: sec checkpoint bridge

Commit:
adds EndoStation

Commit:
rejigs everything around a bit fixes issues people had

Commit:
fixed a gaping whole in space from where i moved secure storage

Commit:
adds cyro to medbay

Commit:
adds a music instrument vending machine

Commit:
fixes most of Helanthius issues 20/50

should fix travis issue

Commit:
adds cuban petes classic beats

Commit:
adds more slogans for cuban pete, also pre custom shuttle

Commit:
adds custom shuttle for Endo

Commit:
addressed Frances issues (few walls etc)

Commit:
fixes crossed issues

Commit:
adds Endostation for Sge

Commit:
round ones bug fixes

Commit:
all issues pointed out minus cameras

Commit:
All Issues up to Super12s on Forums

Commit:
fuck knows, more bug fixes id imagine

Commit:
should fix all space tiles in station, and fixes all forum issues

thank you for all the feedback

Commit:
no pipe errors, found more space floor, sorted out freezer

Commit:
Fixes Power Issues

Commit:
bug fixes

Commit:
fixes whole under hop desk, adds more directions

Commit:
Removes 2 smes swaps another for a fuller smes solving power RS

gives 15-20 mins before start to have power issues with nothing going on

Commit:
adds more teleportation hubs

fixes endos shuttle

Commit:
Adds Medbot and a floor bot

Commit:
adds price tagger to shop and adds disposal to the shop

Blueturbo47 committed:
Signed-off-by: Blue <Blueturbo47@gmail.com>
Nya

Blueturbo47 committed:
Merge pull request #1 from LastCrusader105/ZuluStation

My Legacy

PowerfulBacon commented on code/_globalvars/lists/flavor_misc.dm:
```diff
@@ -171,7 +173,7 @@ GLOBAL_LIST_INIT(TAGGERLOCATIONS, list("Disposals",
 	"Bar", "Kitchen", "Hydroponics", "Janitor Closet","Genetics",
 	"Testing Range", "Toxins", "Dormitories", "Virology",
-	"Xenobiology", "Law Office","Detective's Office"))
+	"Xenobiology", "Law Office","Detective's Office","Plumbing","Morgue"))
```
Adding new things to this is a bad idea, otherwise you should be updating every other maps disposals to allow for this.

Blueturbo47 replied:
Crusader told me that the parcels will just go back to the mailroom and nothing should break.

francinum replied:
It's best to not do this still, as it causes inconsistency with maps that don't have these defined. I can tell you a better way of implementing this if you grab me on discord.

PowerfulBacon reviewed (commented):
The reason the map was abandoned in the first place was due to a large number of flaws that popped up during the testing phase.
There is some significant work that would need to go into this PR and some of the issues I would point out myself
 - Certain areas are too large and need to be made smaller / converted to maints. Areas should not be big for the sake of being big, and if they have empty space should be converted into maints / removed. If you are adding more things to a room then of course it would be bigger, but making a room twice the size makes it twice as empty, not twice as good.
 - Decals are extremely inconsistent since large portions of the map are taken from other existing maps. This creates a lot of confusing areas since it is very strange to see a part of another map (EG deltaEvac) but have it lead to a different location. Additionally there is a lot of disparity between the decal design of areas that have been taken from Kilo station (AI sat) and areas of the map taken from Box / Meta. The 2 different stations use very opposite and extremely different design choices that do not work well together when combined. I would recommend fully removing all decals and redoing them with a consistent design.
 - There are some smaller bugs, see the forum thread for it.

This is an insanely large project to finish, you have been warned.

Blueturbo47 commented:
> * Certain areas are too large and need to be made smaller / converted to maints. Areas should not be big for the sake of being big, and if they have empty space should be converted into maints / removed. If you are adding more things to a room then of course it would be bigger, but making a room twice the size makes it twice as empty, not twice as good.

Any areas you suggest making smaller or bigger I will see what I can do.

> * Decals are extremely inconsistent since large portions of the map are taken from other existing maps. This creates a lot of confusing areas since it is very strange to see a part of another map (EG deltaEvac) but have it lead to a different location. Additionally there is a lot of disparity between the decal design of areas that have been taken from Kilo station (AI sat) and areas of the map taken from Box / Meta. The 2 different stations use very opposite and extremely different design choices that do not work well together when combined. I would recommend fully removing all decals and redoing them with a consistent design.

A lot of the smaller things I can fix any areas I may miss or just not see ping me on discord or something, also is there a way to take a picture of a map in strongdmm to update the photo?

> * There are some smaller bugs, see the forum thread for it.
 
I didn't know there was a forum thread but I will be checking that out now, I've just been in the discord mostly.

Blueturbo47 committed:
Merge pull request #3 from BeeStation/master

Updates myaster

Blueturbo47 committed:
Merge pull request #4 from BeeStation/master

Update bwanch

Blueturbo47 committed:
Merge branch 'EndoStation' of https://github.com/Blueturbo47/BeeStation-Hornet into master

Blueturbo47 committed:
Check Google Doc

Blueturbo47 commented:
Here is my changelog google doc. 
https://docs.google.com/document/d/1WF_-65vnfNWl1z4tqdEBMOQZLGLAg3OhZsN8pg4xlOk/edit?usp=sharing

Blueturbo47 commented:
Cannot post the full map unfortunately because it's above 10mb. 
Here is one missing some carp spawners missing on the outside, also I'm not sure if I added too many.
![RealEndostationunder10](https://user-images.githubusercontent.com/65003500/101730145-8158ab80-3a87-11eb-9b14-5e6fac15bc5d.png)

ToasterBan commented:
when the hands of endo go from crusader to Collin, how has it fallen

Blueturbo47 commented:
> when the hands of endo go from crusader to Collin, how has it fallen

Don't worry the felinid erp room is coming soon :)

Blueturbo47 committed:
Check Google Doc
nya

Blueturbo47 commented:
Here is the updated map, currently I'm out of idea's and I don't see much that needs fixing, still haven't heard back from bacon and crossed still hasn't opened the thread.
![RealEndostationunder10new1](https://user-images.githubusercontent.com/65003500/101802093-0fab4c80-3add-11eb-913f-3a3896a79479.png)

PowerfulBacon commented:
![image](https://user-images.githubusercontent.com/26465327/101811955-a2b2a980-3b12-11eb-8a51-c4f0575f2d0f.png)

Blueturbo47 committed:
A lot of Bacon's changes
Check Google Doc

Blueturbo47 commented:
Here is a new map bacon, I am still working on your suggestions though

- Im not changing the shuttles/ruins outside the station, crusader liked them - Done
- The departures entrance is not a copy says crusader. - Done
- The ai sat looks fine to me, if anything I just want this merged any changes that arent game-breaking I can work on later -Done
- Crusader liked the carpet and its arcade carpet for a reason - Done
- There is an interrogation intercom you just missed it - Done
- I will be making atmos empty enough so it's not weird that it's empty in the future when I have time - Done
- I will be moving the arcade airlock - Done
- I will fix that ugly shuttle by adding some more stuff - Done
- I will add some stuff to atmos space like catwalks - Done
- Decals to xenobio coming soon - Done
- Cant really move toxins bomb testing but I added a paper to tell them where to go - Done
- Decal art coming to chem lab soon - Done
- Intercoms to bridge coming soon - Done

Stuff that isn't done.

- Noticed that seat in sm room I'm gonna fix that
- Suggestions to fill up robotics lab would be nice

Here is the new map of this commit though
![RealEndostationunder10new2](https://user-images.githubusercontent.com/65003500/101852987-dac6e600-3b2c-11eb-83f0-42902df228b4.png)

Blueturbo47 committed:
Added more bridge radio's, how do i fix this error?

Blueturbo47 committed:
Fixes that error
uwu

Blueturbo47 committed:
Updates atmos to make it a lot better
Is that more filled bacon?

Blueturbo47 commented:
Is that enough space @PowerfulBacon ?
<img width="527" alt="For Bacon" src="https://user-images.githubusercontent.com/65003500/101959258-fab0e500-3bd2-11eb-8abe-3701879d71b2.PNG">

Blueturbo47 committed:
Updated shuttle and depatures room
my internet hasn't cut out in a bit and i find that strange, making commits while i can

Blueturbo47 committed:
Ive taken in **everything** bacon said and now this is the final touches, anyone else wanna input?
Google Doc - https://docs.google.com/document/d/1WF_-65vnfNWl1z4tqdEBMOQZLGLAg3OhZsN8pg4xlOk/edit?usp=sharing
Discord - https://discord.gg/BNsjUTEQXX

Blueturbo47 commented:
Here is the map after I've taken in everything bacon said, should be good to merge unless anyone wants anything else added.

![RealEndostationunder10baconfinish](https://user-images.githubusercontent.com/65003500/101967193-a3b60a80-3be8-11eb-9fd7-9811efea4c4f.png)

Blueturbo47 committed:
Recolors a pipe in atmos

PowerfulBacon commented:
Atmos feels even more empty now, since you removed the only thing in the middle making it not empty. A large majority of the map's rooms could be reduced in size, they are currently big for the sake of being big and contain nothing extra in the new space.

PowerfulBacon commented:
https://wiki.beestation13.com/view/Guide_to_mapping

Check these imporantly
```
    Are floors with or without air, as they should be? (regular or airless)
    Does the area have an APC?
    Does the area have an Air Alarm?
    Does the area have a Request Console?
    Does the area have lights?
    Does the area have a light switch?
    Does the area have enough intercoms?
    Does the area have enough security cameras? (Use the verbs under Mapping for help)
    Is the area connected to the scrubbers air loop?
    Is the area connected to the vent air loop? (vent pumps)
    Is everything wired properly?
    Does the area have a fire alarm and firedoors?
    Do all pod doors work properly?
    Are accesses set properly on doors, pod buttons, etc.
    Are all items placed properly? (not below vents, scrubbers, tables)
    Does the disposal system work properly from all the disposal units in this room and all the units, the pipes of which pass through this room?
    Check for any misplaced or stacked piece of pipe (air and disposal)
    Check for any misplaced or stacked piece of wire
    Identify how hard it is to break into the area and where the weak points are, and balance the area accordingly (eg. the Vault should be made of reinforced structures and electrified windows, the Kitchen should not)
    Check if the area has too much empty space. If so, make it smaller and replace the rest with maintenance tunnels
    Are there any indestructible turfs where they shouldn't be?
```

FortSumter commented:
To be fair bacon, Pubby still isn't fully fastmos compatible. If you plasma flood (science in particular) you can't siphon a lot of rooms due to the only scrubber being behind the thrice-accursed new firelocks.

As far as this map goes I like it. I don't mind big rooms so much because it gives the players some space to do whatever with. Some tables with stuff on them might be nice just to fill up some space if you want.

If you're still looking for suggestions on what to give robotics, there's a lot of things they can use round start, such as the makings for different kinds of bots, maybe their own autolathe, batteries, extra cable coils, and if you want to be the mapper MVP, Circuitry equipment. Pending permission, you could also add a drone fabricator to make drones, which would be extremely useful for repairing a station of this size. Perhaps even add an unfinished Ripley mech, because why not?

Blueturbo47 commented:
Can I get this merged or closed? I fucked up my master branch and need to create a new fork because I am a bad coder but I don't want to open a new or just because I fucked up my master branch.
I just want the map to work, for now, anything that I miss or needs to be changed I will change it or you can make a pr to change it

PowerfulBacon commented:
> 
> 
> Can I get this merged or closed? I fucked up my master branch and need to create a new fork because I am a bad coder but I don't want to open a new or just because I fucked up my master branch.
> I just want the map to work, for now, anything that I miss or needs to be changed I will change it or you can make a pr to change it

Sorry, didn't see this until now since I've been pretty busy recently. I'll close, since I don't think the map is in a state of high enough quality to be an activate rotation map at the moment and has a few bits that need changing to be more in line with the current maps in rotation.
